core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -1416,8 +1426,10 @@ public void checkIfSchemaWithIdExist(int id, Schema schema)
         schemaCopy.setSubject(existingSchema.getSubject());
         schemaCopy.setVersion(existingSchema.getVersion());
         if (!existingSchema.equals(schemaCopy)) {
+          String context = QualifiedSubject.qualifiedContextFor(tenant(), schema.getSubject());
           throw new OperationNotPermittedException(
-              String.format("Overwrite new schema with id %s is not permitted.", id)
+              String.format("Overwrite new schema with id %s is not permitted."
+              +  "(context: %s)", id, context)
```

Comment:
Similar change as mentioned previously
```suggestion
              String.format("Overwrite new schema with id %s in context '%s' is not permitted.", 
              id, context)
```